services/provider/client/client.go

```diff
@@ -2,49 +2,55 @@ package client
 
 import (
 	"context"
+	"fmt"
 	"time"
 
+	"github.com/red-hat-storage/ocs-operator/services/provider/common"
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
 	"google.golang.org/grpc"
-	"k8s.io/klog"
 )
 
-type ConsumerClient struct {
+type OCSProviderClient struct {
 	Client  pb.OCSProviderClient
 	timeout time.Duration
 }
 
-// NewConsumerClient creates a ConsumerClient to talk to OCS consumer server
-func NewConsumerClient(cc *grpc.ClientConn, timeout time.Duration) *ConsumerClient {
-	return &ConsumerClient{Client: pb.NewOCSProviderClient(cc), timeout: timeout}
+// NewProviderClient creates a client to talk to OCS provider server
+func NewProviderClient(cc *grpc.ClientConn, timeout time.Duration) *OCSProviderClient {
+	return &OCSProviderClient{Client: pb.NewOCSProviderClient(cc), timeout: timeout}
+}
+
+// NewMockProviderClient creates a client to mock the OCS provider server api calls
+func NewMockProviderClient(errorFlow common.MockErrorFlow) *OCSProviderClient {
```

Comment:
Why do we need mock provider client?

Reply:
Actually, we are creating this for testing purposes as our API server and controller work is blocked for now.